### ISSUE TITLE
Doppelganger capturing size-changing dice

### ISSUE THREAD

**Issue by irilyth** [buttonmen-dev/buttonmen#2223]
When a Doppelganger die captures a Mighty or Weak die in a Power Attack, the Doppelganger die changes into the target die type, and then changes into a larger or smaller die when it's rerolled. When a Doppelganger die captures a Turbo Swing die in a Power Attack, the capturing player seems not to have the option to use Turbo to change it into a larger or smaller die when it's rerolled. It seems like Mood Swing works like Turbo in this way. Is that inconsistency intentional? It seemed surprising to me.

**Comment by blackshadowshade** [comment]
Yes, it's intentional. It's all about timing. Currently, we set turbo swing values when the attack is being specified, but with a doppelganger die, the UI doesn't know that the doppelganger die will become a turbo die.

Unless you're suggesting that perhaps the mighty/weak shouldn't trigger after doppelganger. That would seem reasonable if you're looking for consistency.

**Comment by irilyth** [comment]
Hm, couple things.

By "intentional", I meant more like "from a gameplay perspective", as opposed to "a technical necessity".

Why doesn't the UI know that the Doppelganger die will become a Turbo die? That logic is only in the backend? It seems like it *could* know that -- all the information there is available to it, right? It's not like it depends on the *result* of the attack, or something.

This occurred to me again because a Warrior Turbo Swing die *does* get to change size when it comes into play.

I think from a gameplay perspective, it'd be better if a Doppelganger die capturing a Turbo Swing die did get to change size.
